core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -105,16 +105,31 @@ class ReplicaFetcherThread(name:String,
       leaderEndOffset
     } else {
       /**
-       * The follower could have been down for a long time and when it starts up, its end offset could be smaller than the leader's
+       * If the leader's log end offset is greater than follower's log end offset, there are two possibilities:
+       * 1. The follower could have been down for a long time and when it starts up, its end offset could be smaller than the leader's
        * start offset because the leader has deleted old logs (log.logEndOffset < leaderStartOffset).
+       * 2. When leader migration occurs, there is a slight chance that the old leader's high watermark is greater than
+       * the new leader's log end offset, so an OffsetOutOfRangeException was thrown. After that some more messages are
+       * produced to the leader. When the follower was trying to handle the exception and query the log end offset of
```

Comment:
Can you reword the second scenario and clarify? Especially the part where "However, there is some time between the follower receives the OffsetOutOfRange exception and starts to handle it. That means when the follower got the exception first time, OST might be greater than leader's LEO. But when the follower starts to handle the exception and query the LEO again, OST is already smaller than leader's LEO - because the leader might already got some new messages appended, the offset perhaps is no longer too big."

It is non-intuitive and understanding this is important for understanding the code and our logic.